### ISSUE TITLE
eslint erroneously complains about arrow function

### ISSUE THREAD

**Issue by tromey** [eslint/eslint#2616]
Consider this code:

```
const steppingHookState = {
    pauseAndRespond: (aFrame, onPacket=(k)=>k) => {
        return this._pauseAndRespond(aFrame, { type: "resumeLimit" }, onPacket);
    }
};
```

When I run eslint on this (using the "all off" config from https://bugzilla.mozilla.org/show_bug.cgi?id=887895), I get:

```
pokyo. eslint --no-eslintrc -c /tmp/eslintrc-all-off.json /tmp/q.js

/tmp/q.js
  2:48  error  Unexpected token =>

✖ 1 problem (1 error, 0 warnings)
```

However, I think this code is actually correct, and eslint is incorrect in reporting an error.

A workaround is to reword the default argument from "(k)=>k" to "k=>k".  This has the same meaning but suppresses the eslint error.

**Comment by xjamundx** [comment]
If it's a bug it's likely an espree bug. See this in my node REPL:

```js
require('espree').parse('(a, b=(c)=>c) => {}', {ecmaFeatures: { arrowFunctions: true }})
```

Can you open up an issue with espree?
https://github.com/eslint/espree/issues

**Comment by xjamundx** [comment]
q: are you essentially using `(k)=>k` as a noop (empty function)?

**Comment by tromey** [comment]
> Can you open up an issue with espree?

https://github.com/eslint/espree/issues/145

> q: are you essentially using `(k)=>k` as a noop (empty function)?

It's the identity function.

I don't know the history or intent of the code in question -- it could certainly be clearer -- I was just trying to help convert some parts of the tree to be parseable by eslint and this came up as a seemingly erroneous error.  (The only one I found btw.)

**Comment by xjamundx** [comment]
Thanks @tromey we'll get this fixed!

**Comment by xjamundx** [comment]
We can probably close this now.

**Comment by ilyavolodin** [comment]
Closing for https://github.com/eslint/espree/issues/145
